crates/worker/src/partition_processor_manager.rs

```diff
@@ -684,68 +759,75 @@ impl<T: TransportConnect> PartitionProcessorManager<T> {
-                        if let Err(err) = handle.await {
-                            warn!("Partition processor crashed while shutting down: {err}");
-                        }
-                    }
-                } else {
-                    debug!("No running partition processor. Ignoring stop command.");
+                if let Some(status) = self.running_partition_processors.remove(&partition_id) {
+                    status.stop(&self.task_center).await;
                 }
             }
-            ProcessorCommand::Follower => {
-                if let Some(state) = self.running_partition_processors.get_mut(&partition_id) {
-                    // if we error here, then the system is shutting down
-                    state.step_down()?;
-                } else if let Some(partition_key_range) = partition_table
-                    .get_partition(&partition_id)
-                    .map(|partition| &partition.key_range)
-                {
-                    self.start_partition_processor(
-                        partition_id,
-                        partition_key_range,
-                        RunMode::Follower,
-                    )
-                    .await?;
+            ProcessorCommand::Follower | ProcessorCommand::Leader => {
+                let run_mode = if control_processor.command == ProcessorCommand::Leader {
+                    RunMode::Leader
+                } else if control_processor.command == ProcessorCommand::Follower {
+                    RunMode::Follower
                 } else {
-                    debug!("Unknown partition id '{partition_id}'. Ignoring follower command.");
-                }
-            }
-            ProcessorCommand::Leader => {
-                if let Some(state) = self.running_partition_processors.get_mut(&partition_id) {
-                    state
-                        .run_for_leader(
-                            self.metadata_store_client.clone(),
-                            self.metadata.my_node_id(),
-                        )
-                        .await?;
+                    unreachable!()
+                };
+
+                if let Some(status) = self.running_partition_processors.get_mut(&partition_id) {
+                    if let ProcessorStatus::Started(state) = status {
+                        // if we error here, then the system is shutting down
+                        if run_mode == RunMode::Follower {
+                            state.step_down()?;
+                        } else if run_mode == RunMode::Leader {
+                            state
+                                .run_for_leader(
+                                    self.metadata_store_client.clone(),
+                                    self.metadata.my_node_id(),
+                                )
+                                .await?;
+                        }
+                    }
+                    // otherwise if it's still starting we can't control it yet
```

Comment:
Maybe add a debug log statement that we are ignoring this command because the PP is still starting.